PyRoute/DeltaDebug.py

```diff
@@ -0,0 +1,172 @@
+                        help='Dump internal data structures as json for later further processing ')
+
+    delta = parser.add_argument_group('Debugging', 'Parameters for the delta-debugging procedure itself')
+    delta.add_argument('--input', default='sectors', help='input directory for sectors')
+    delta.add_argument('--sectors', default=None, help='file with list of sector names to process')
+    delta.add_argument('sector', nargs='*', help='T5SS sector file(s) to process')
+    delta.add_argument('--output-dir', dest="outputdir",
+                       help='Output folder to place any maps generated during minimisation.')
+    delta.add_argument('--min-dir', dest="mindir",
+                       help='Output folder to place minimised sector files generated during minimisation.')
+    delta.add_argument('--interesting-line', dest="interestingline", default=None,
+                       help="Line required to classify run output as interesting")
+    delta.add_argument('--interesting-type', dest="interestingtype", default=None,
+                       help="Exception type required to classify run output as interesting")
+    delta.add_argument('--two-minimise', dest="two_min", default=False, action='store_true',
+                       help="Try all pairs of star lines to see if any can be removed")
+    delta.add_argument('--no-sector', dest="run_sector", default=True, action='store_false',
+                       help="Skip sector-level reduction")
+    delta.add_argument('--no-subsector', dest="run_subsector", default=True, action='store_false',
+                       help="Skip subsector-level reduction")
+    delta.add_argument('--no-line', dest="run_line", default=True, action='store_false',
+                       help="Skip line-level reduction.  At least one of sector, subsector, line and two-minimisation must be selected")
+    delta.add_argument('--assume-interesting', dest="run_init", default=True, action='store_false',
+                       help="Assume initial input is interesting.")
+
+    args = parser.parse_args()
+
+    # sanity check run arguments
+    if not (args.two_min or args.run_sector or args.run_subsector or args.run_line):
+        raise ValueError("Must select at least one reduction pass to run")
+
+    galaxy = Galaxy(args.btn, args.max_jump, args.route_btn)
+    galaxy.output_path = args.output
+
+    sectors_list = args.sector
+    if args.sectors is not None:
+        sectors_list.extend(get_sectors(args.sectors, args.input))
+
+    delta = DeltaDictionary()
+    for sector_file in sectors_list:
+        sector = SectorDictionary.load_traveller_map_file(sector_file)
+        delta[sector.name] = sector
+
+    logger.error("%s sectors read" % len(delta))
+
+    # spin up reducer
+    reducer = DeltaReduce(delta, args, args.interestingline, args.interestingtype)
+
+    # check original input is interesting
+    if args.run_init:
+        reducer.is_initial_state_interesting()
```

Comment:
So far, assume-interesting has been helpful in finding bugs in delta debugging itself.  As I've found it useful, I've added it for others to use.